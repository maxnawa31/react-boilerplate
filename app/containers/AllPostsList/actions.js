/*
 *
 * AllPostsList actions
 *
 */

import {
  LOAD_POSTS,
  LOAD_POSTS_SUCCESS,
  LOAD_POSTS_ERROR,
  DEFAULT_ACTION,
} from './constants';

export function loadPosts() {
  return {
    type: LOAD_POSTS,
  };
}

export function loadPostsSuccess(posts) {
  return {
    type: LOAD_POSTS_SUCCESS,
    posts,
  };
}

export function loadPostsError(error) {
  return {
    type: LOAD_POSTS_ERROR,
    error,
  };
}
export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
