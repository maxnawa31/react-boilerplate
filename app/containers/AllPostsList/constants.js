/*
 *
 * AllPostsList constants
 *
 */

export const DEFAULT_ACTION = 'app/AllPostsList/DEFAULT_ACTION';
export const LOAD_POSTS = 'app/AllPostsList/LOAD_POSTS';
export const LOAD_POSTS_SUCCESS = 'app/AllPostsList/LOAD_POSTS_SUCCESS';
export const LOAD_POSTS_ERROR = 'app/AllPostsList/LOAD_POSTS_ERROR';
export const SELECT_POST = 'app/AllPostsList/SELECT_POST';
