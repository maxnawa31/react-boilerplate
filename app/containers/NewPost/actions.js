/*
 *
 * NewPost actions
 *
 */

import {
  NEW_POST,
  NEW_POST_SUCCESS,
  NEW_POST_ERROR,
  DEFAULT_ACTION,
} from './constants';

export function newPost(newPostData) {
  return {
    type: NEW_POST,
    newPost: newPostData,
  };
}

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function newPostSuccess() {
  return {
    type: NEW_POST_SUCCESS,
  };
}

export function newPostError() {
  return {
    type: NEW_POST_ERROR,
  };
}
