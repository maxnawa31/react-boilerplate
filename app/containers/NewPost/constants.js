/*
 *
 * NewPost constants
 *
 */

export const NEW_POST = 'app/NewPost/NEW_POST';
export const NEW_POST_SUCCESS = 'app/NewPost/NEW_POST_SUCCESS';
export const NEW_POST_ERROR = 'app/NewPost/NEW_POST_ERROR';
