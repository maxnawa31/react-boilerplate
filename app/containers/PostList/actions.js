import {
  LOAD_POSTS,
  LOAD_POSTS_SUCCESS,
  LOAD_POSTS_ERROR,
  SELECT_POST,
} from './constants';

export function loadPosts(url) {
  console.log('inside LOAD_POSTS action', url);
  return {
    type: LOAD_POSTS,
  };
}
export function loadPostsSuccess() {
  return {
    type: LOAD_POSTS_SUCCESS,
  };
}
export function loadPostsError() {
  return {
    type: LOAD_POSTS_ERROR,
  };
}

export function selectPost() {
  return {
    type: SELECT_POST,
  };
}
