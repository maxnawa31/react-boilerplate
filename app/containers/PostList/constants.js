export const LOAD_POSTS = 'LOAD_POSTS';
export const LOAD_POSTS_SUCCESS = 'LOAD_POSTS_SUCCESS';
export const LOAD_POSTS_ERROR = 'LOAD_POSTS_ERROR';
export const SELECT_POST = 'SELECT_POST';
