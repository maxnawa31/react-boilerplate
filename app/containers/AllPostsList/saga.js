import { put, takeLatest } from 'redux-saga/effects';
import axios from 'axios';
import { loadPostsError, loadPostsSuccess } from './actions';
import { LOAD_POSTS } from './constants';

function* fetchPostsAsync() {
  try {
    const response = yield axios.get('http://localhost:5000');
    yield put(loadPostsSuccess(response.data));
  } catch (err) {
    yield put(loadPostsError(err));
  }
}

export default function* watchLoadPostsAsync() {
  yield takeLatest(LOAD_POSTS, fetchPostsAsync);
}

// export default function* postsData() {
//   yield takeLatest(LOAD_POSTS, watchLoadPostsAsync);
// }
