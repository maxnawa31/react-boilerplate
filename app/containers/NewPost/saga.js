import { call, put, takeEvery } from 'redux-saga/effects';
import axios from 'axios';
import { NEW_POST_SUCCESS, NEW_POST, NEW_POST_ERROR } from './constants';

// Individual exports for testing

function* makeNewPostAsync(action) {
  try {
    const postData = action.newPost;
    yield call(axios.post, 'http://localhost:5000/', {
      text: postData,
    });
    yield put({ type: NEW_POST_SUCCESS });
  } catch (error) {
    yield put({ type: NEW_POST_ERROR, error });
  }
}

export default function* watchNewPostAsync() {
  yield takeEvery(NEW_POST, makeNewPostAsync);
}
