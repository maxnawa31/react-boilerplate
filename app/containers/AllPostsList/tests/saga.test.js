/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { take, call, put, select, takeLatest } from 'redux-saga/effects';
import { fetchPostsAsync, watchLoadPostsAsync } from '../saga';
import { loadPostsError, loadPostsSuccess } from '../actions';
import { LOAD_POSTS_ERROR, LOAD_POSTS_SUCCESS, LOAD_POSTS } from '../constants';
// const generator = defaultSaga();

describe('defaultSaga Saga', () => {
  let getDataGenerator;
  beforeEach(() => {
    getDataGenerator = fetchPostsAsync();
    expect(getDataGenerator).toMatchSnapshot();

    const apiCall = getDataGenerator.next().value;
    expect(apiCall).toMatchSnapshot();
  });
  it('should dispatch data if api call is successful', () => {
    const response = {
      data: [
        {
          created_at: '2018-10-06T00:33:39.788Z',
          id: 235,
          text: 'sdfsfs',
        },
        {
          created_at: '2018-10-06T00:36:49.251Z',
          id: 236,
          text: 'sfsdfs',
        },
      ],
    };
    const putDescriptor = getDataGenerator.next(response).value;

    expect(putDescriptor).toMatchSnapshot();
    expect(putDescriptor).toEqual(put(loadPostsSuccess(response.data)));
  });
  it('should call loadPostError if api call is not successful', () => {
    const response = new Error('There was an error');
    const putDescriptor = getDataGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loadPostsError(response)));
  });
});

describe('watchLoadPostsAsync Saga', () => {
  const loadPostWatcher = watchLoadPostsAsync();
  it('should start task to watch for LOAD_POSTS action', () => {
    const takeLatestDescriptor = loadPostWatcher.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(LOAD_POSTS, fetchPostsAsync),
    );
  });
});
