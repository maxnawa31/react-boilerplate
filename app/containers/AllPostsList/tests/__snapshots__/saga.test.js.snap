// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultSaga Saga should call loadPostError if api call is not successful 1`] = `
GeneratorFunctionPrototype {
  "_invoke": [Function],
}
`;

exports[`defaultSaga Saga should call loadPostError if api call is not successful 2`] = `Promise {}`;

exports[`defaultSaga Saga should dispatch data if api call is successful 1`] = `
GeneratorFunctionPrototype {
  "_invoke": [Function],
}
`;

exports[`defaultSaga Saga should dispatch data if api call is successful 2`] = `Promise {}`;

exports[`defaultSaga Saga should dispatch data if api call is successful 3`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "posts": Array [
        Object {
          "created_at": "2018-10-06T00:33:39.788Z",
          "id": 235,
          "text": "sdfsfs",
        },
        Object {
          "created_at": "2018-10-06T00:36:49.251Z",
          "id": 236,
          "text": "sfsdfs",
        },
      ],
      "type": "app/AllPostsList/LOAD_POSTS_SUCCESS",
    },
    "channel": null,
  },
}
`;
